tests/unittests/CoreGraphics.drawing/ImageComparison.cpp

```diff
@@ -18,91 +18,203 @@
 #include "ImageHelpers.h"
 
 #include <CoreGraphics/CGGeometry.h>
+#include <algorithm>
+#include <vector>
 
-struct bgraPixel {
-    uint8_t b, g, r, a;
+struct Pixel {
+    uint8_t r, g, b, a;
+    bool operator==(const Pixel& o) const {
+        return r == o.r && g == o.g && g == o.g && a == o.a;
+    }
 };
 
-struct rgbaPixel {
-    uint8_t r, g, b, a;
+struct ImagePixelAccess {
+private:
+    woc::unique_cf<CGImageRef> _image;
+    woc::unique_cf<CFDataRef> _cfData;
+    const uint8_t* _data; // view into cfData
+    CGBitmapInfo _byteOrder;
+    CGImageAlphaInfo _alphaInfo;
+    bool _premultiplied;
+    bool _alphaFirst;
+    bool _alphaSkip;
+
+    static uint8_t unpremultiply(uint8_t val, uint8_t alpha) {
+        if (alpha == 0) {
+            return 0;
+        }
+        float fval = (float)val / (float(alpha) / 255.f);
```

Comment:
nit: using the one float like a cast and other like a function makes this mighty uncomfortable to parse, pick one pls